File: webapp/src/app/users/model/user.model.ts
import {UserRole} from './user-role.model';

export class User {

  constructor(public id: number,
              public username: string,
              public firstName: string,
              public lastName: string,
              public userRole: UserRole,
              public isActive: boolean,
              public email: string
  ) {
  }
}

File: src/main/java/ro/esolutions/eipl/models/UserModel.java
package ro.esolutions.eipl.models;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
public class UserModel implements HasUsernameAndEmail {

    private Long id;

    @Size(min = 2)
    private String username;

    @Size(min = 2)
    private String firstName;

    @Size(min = 2)
    private String lastName;

    @NotNull
    private UserRole userRole;

    @NotNull
    private Boolean isActive;

    @NotNull
    @Size(min = 5)
    private String email;
}
